### ISSUE TITLE
Observable.timeout Zone interference

### ISSUE THREAD

**Issue by tiagoroldao** [angular/angular#13029]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
When using `Observable.timeout` the inherent subscription is caught by the angularZone, and isn't cleaned up after the original source is closed. 
As a result, if a timeout of 10s is in an Observable chain, the app is marked untestable during that period, regardless of wether or not the Observable has resolved and finished before that time.

**Expected behavior**
After the original source closes, application should be stable regardless of timeout status.

**Minimal reproduction of the problem with instructions**
Plunkr showing problem: https://plnkr.co/edit/3QA9lGuFBq5yuAFdJ32E?p=preview

Tick `Subscription with Late Timeout` to see it happening, in a subscription for a 1 second timer, and a timeout of 5 seconds

**What is the motivation / use case for changing the behavior?**
- Timeout is used frequently for asynchronous calls. 
- Behaviour prevents proper testing due to protractor timeouts, and isn't easily solvable through `runOutsideAngular` as the source Observable is often intended to run within the app

**Please tell us about your environment:**
MacOS, tested in Chrome

* **Angular version:** 2.2.1

* **Browser:** presumed all (only tested in Chrome, not browser-specific)
 
* **Language:** TypeScript

* **Notes:**
This is due to how RxJS currently works - this operator starts their own subscription, that cancels the original source if it is not closed when it resolves. 
Tested with the angular required version, but upon looking at `rxjs`'s source, is still an issue with the latest RC candidate.

**Comment by gdi2290** [comment]
please follow https://github.com/ReactiveX/rxjs/issues/2134 for this issue

**Comment by tiagoroldao** [comment]
@gdi2290 cheers! Does it make sense to keep tabs on the issue from here? Otherwise I'll close this issue.

**Comment by gdi2290** [comment]
you can close this issue as it's a rxjs issue

**Comment by gdi2290** [comment]
actually, this won't be fixed until rxjs releases their fix and angular updates their peer dep but I'll mention this is another issue

**Comment by tiagoroldao** [comment]
Reopening this as latest angular user rxjs@^5.0.2, but problem seems to persist. 
@gdi2290 I can close this again, but as it is relevant thought best to notify.

**Comment by gdi2290** [comment]
yeah still waiting on https://github.com/ReactiveX/rxjs/issues/2134

**Comment by gdi2290** [comment]
this is fixed in rxjs version 5.2.0+

**Comment by mhevery** [comment]
I believe this is intended behavior. `Observable.timeout` is just a fancy way to do `setTimeout` and `setTimeout` has exactly this behavior. So I will argue this is working as intended.

**Comment by tiagoroldao** [comment]
I don't know if this has since been resolved, but the comparison between Observable.timeout and setTimeout seems incorrect to me:
http://reactivex.io/documentation/operators/timeout.html

Timeout forces a failure on an observable. It stand to reason that closing the observable (making it NOT fail after the timeout set) should also close timeout hook. Again, thought it worth mentioning, but if angular now depends on rxjs 5.2 or above, this may be closed anyway.
